core/workspace.js

```diff
@@ -40,7 +41,7 @@ Blockly.Workspace = function(opt_options) {
   this.id = Blockly.utils.genUid();
   Blockly.Workspace.WorkspaceDB_[this.id] = this;
   /** @type {!Blockly.Options} */
-  this.options = opt_options || /** @type {!Blockly.Options} */ ({});
+  this.options = opt_options || new Blockly.Options({});
```

Comment:
Not sure this fixes the issue. This might use a better default if no options are passed, but what if a ``{}`` is passed in. It wouldn't go through Blockly.Options, and thus ``options.oneBasedIndex`` would be set to false.

Reply:
I've updated the logic to be the same as in inject.

Reply:
Changed back after noticing there is `Blockly.Options` and a `Blockly.BlocklyOptions`,
I think this makes sense here because if someone was to pass an empty object, it would actually be the wrong type and I think its fine that it isn't parsed.
This fixes the inconsistency between the constructor with no option passed and inject with no option passed.

Reply:
Okay that makes sense. In which case, I would update any of our tests that pass in an object in the constructor to instead wrap that object with a Blockly.Options object.

Reply:
Test have been updated.